Provide easy-to-use ItchySats docker guide (#1)
itchysats/docker-itchysats: README.md

# docker-itchysats

A repo facilitating running ItchySats inside a [Docker][https://www.docker.com/what-docker] container on a x86-64 (e.g. macOS, Linux, Windows) or ARMv7 system.
Docker website provides extensive tutorials on Docker installation, please refer there if you are new to Docker.

> The following guide assumes that you have Docker running, having [docker-compose][https://docs.docker.com/compose/install/] installed is also recommended.

Reviewer: Lol. 
If you have to explain someone Docker in a readme about how to run ItchySats in a docker container  I don't think they will get far 😅 

I would prefer an introduction to ItchySats instead.

Author: well, this is a guide to running itchysats in a docker container - introduction to itchysats is linked at the bottom.
 I will add a link to `ItchySats`, so anyone confused can quickly jump there if they're confused (if they don't know, they should be on a different page :D)

Reviewer: > well, this is a guide to running itchysats in a docker container - introduction to itchysats is linked at the bottom.

The question is, how do you expect people to find this repo? 

- If it pops up on Google, I'd expect to see an intro into ItchySats first. 
- If you link to it from a blog post/twitter thread explaining how to run ItchySats in a docker container, I would not expect anyone to click on the link who does not know Docker. 

Eitherway, I'm fine with having this here, it's just my two Sats :) 

> if they don't know, they should be on a different page :D

Totally agree with this :)

Author: thank you for the comments, I have reworked the readme to briefly describe ItchySats first and provide more links to the main repo. Please have a look it if looks more approachable.

as for Docker proficiency - I opted for the link so we would not exclude people who would like to try running it on their system without prior experience with Docker. it doesn't matter for the pros, they will just skip to the next sentence. :)

Reviewer: thanks :)

# Quickstart

> ## Testnet

Reviewer 2: I would put `Mainnet` first. I think it should be more present than `Testnet`.

Author: why? why not allow people try it out of testnet first? I find it important to have it this way for the demo purposes and figuring out what the application can do, before any money is used?

If you just want to check out the application, and want to try out a "demo" version, where you don't trade with real money, then this config is for you.

```
docker-compose project-directory=testnet up -d
```

## Mainnet

If you're ready to trade with real money, use the following command:

```

> docker-compose project-directory=testnet up -d

Reviewer 2: ```suggestion
docker-compose project-directory=mainnet up -d
```

I suppose ;)

> docker-compose project-directory=testnet up -d

Reviewer 2: Can't we default to `mainnet` so that we don't need this for the `mainnet` setup? Would be more convenient.

Author: again, I opted for ease of onboarding and trying it out. If someone copy-pastes the code, it's just as easy (and if they want to be serious, they'll just copy the docker-compose.yml file).

Author: also, if anyone goes inside the mainet/testnet subdir, they can just call `docker-compose up` (I fully suspect that anyone who knows a bit of docker go this way IMHO). The guide is just to kickstart people who are new to it...

```

## User interface

Docker command starts an ItchySats service running in the background.

> The service can be controlled via the provided web UI accessible on [https://localhost:8000][https://localhost:8000]

Reviewer: ```suggestion
The service can be controlled via the provided web UI accessible on [https://localhost:8000][https://localhost:8000].
```

> Web UI requires authentication via password generated from your `taker_seed`.

Reviewer: You can expect people don't know what you mean with `taker_seed`.

Author: good point. looking at the main repo, we lack a clear explanation there either - I'll add it to FAQ and link it there.

Author: https://github.com/itchysats/itchysats/pull/2210 (will link it here when FAQ gets merged)

### Credentials

Credentials can be seen in Docker logs after typing in the following command (example for `mainnet`):

```
docker-compose project-directory=mainnet logs
```

It is strongly recommended to not store ItchySats logs outside your machine without filtering the credentials out.

## Stopping ItchySats

On the example of mainnet:

```
docker-compose project-directory=mainnet down
```

# How to use ItchySats?

You can find more information about ItchySats (along with complete source code) on the main github [repo][https://github.com/itchysats/itchysats].
[FAQ][https://github.com/itchysats/itchysats/blob/master/FAQ.md] might be another good source of knowledge if you get stuck.

# Caveats

> - at this stage there is no automated backup of the wallet seed and the database (please backup up 'data' directory inside `testnet` or `mainnet` and store it securely)

Reviewer 2: ```suggestion
- at this stage there is no automated backup of the wallet seed and the database (please backup up `taker_seed` inside `testnet` or `mainnet` and store it securely)
```

I think they way you configure `testnet` and `mainnet` to be mounted directly, there is no  `data` directory. Or am I missing something?

Author: these is a `data` directory inside mainnet and testnet in this config. You're welcome to try :)
